doc/source/cluster/kubernetes/examples/gpu-training-example.md

```diff
@@ -0,0 +1,149 @@
+kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=v0.3.0&timeout=90s"
+
+# Create a Ray cluster
+kubectl apply -f https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/kubernetes/configs/ray-cluster.gpu.yaml
+
+# Set up port-forwarding
+kubectl port-forward services/raycluster-head-svc 8265:8265
+
+# Test the cluster
+ray job submit --address http://localhost:8265 -- python -c "import ray; ray.init(); print(ray.cluster_resources())"
+
+# Step 3: Run the PyTorch image training benchmark.
+pip3 install -U "ray[default]"
+
+# Download the Python script
+curl https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/doc_code/pytorch_training_e2e_submit.py -o pytorch_training_e2e_submit.py
+
+# Submit the training job to your ray cluster
+python3 pytorch_training_e2e_submit.py
+
+# Use the following command to follow this Job's logs:
+# Substitute the Ray Job's submission id.
+ray job logs 'raysubmit_xxxxxxxxxxxxxxxx' --follow
+```
+
+# Step 1: Setup a Kubernetes cluster on GCP.
+In this section, we set up a Kubernetes cluster with CPU and GPU node pools. These instructions are for GCP, but a similar setup would work for any major cloud provider.
+
+```shell
+# Step 1: Setup a Kubernetes cluster on GCP.
+# e2-standard-8 => 8 vCPU; 32 GB RAM
+gcloud container clusters create gpu-cluster-1 \
+    --num-nodes=1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+    --zone=us-central1-c --machine-type e2-standard-8
+
+# Create a node-pool for GPU
+# n1-standard-8 => 8 vCPU; 30 GB RAM
+gcloud container node-pools create gpu-node-pool \
+  --accelerator type=nvidia-tesla-t4,count=1 \
+  --zone us-central1-c --cluster gpu-cluster-1 \
+  --num-nodes 1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+  --machine-type n1-standard-8
+
+# Install NVIDIA GPU device driver
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+```
+
+If you have an existing Kubernetes cluster with GPU, you can ignore this step. In addition, it is not necessary
+to run this example with a cluster having that much RAM (>30GB per node in the above commands). Feel free to update
+the option `machine-type` and the resource requirements in `ray-cluster.gpu.yaml`.
+
+First, we create a Kubernetes cluster `gpu-cluster-1` with a node (`e2-standard-8`: 8 vCPU; 32 GB RAM). Second,
+we add a new node (`n1-standard-8`: 8 vCPU; 30 GB RAM) with a GPU (`nvidia-tesla-t4`) to the cluster.
+
+# Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+
+:::{note}
+To learn the basics of Ray on Kubernetes, we recommend taking a look
+at the {ref}`introductory guide <kuberay-quickstart>` first.
+:::
+
+If you are new to Kubernetes and you are planning to deploy Ray workloads on a managed
+Kubernetes service, we recommend taking a look at this {ref}`introductory guide <kuberay-k8s-setup>`
+first.
```

Comment:
The above sentence should probably go in step 1, since that's where we set up the K8s cluster on managed K8s service.

Reply:
Updated.